src/app/components/core/page/courselist/courselist.component.html

```diff
@@ -14,7 +14,7 @@
         <h2>Video course {{course.id}}. {{course.heading | uppercase}} &nbsp; <span [ngClass]="course.done ? 'yellow-star' : 'transparent-star'">&#9733;</span></h2>
         <p>{{course.text}}</p>
         <div class="item-nav pull-right">
-          <button class="btn btn-lg btn-primary">&#9998; Edit</button>&nbsp;&nbsp;
+          <button class="btn btn-lg btn-primary" [routerLink] ="['/courses/new-page', 'course.id']">&#9998; Edit</button>&nbsp;&nbsp;
```

Comment:
You're currently sending course id as a string, so the route will be broken

Reply:
> 

do you mean - to change a value of  idCourse: number; (courselist.component.ts) to "string"? (to avoid broking the route)